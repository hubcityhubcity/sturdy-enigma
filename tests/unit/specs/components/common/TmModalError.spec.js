import Vuex from "vuex"
import { shallowMount, createLocalVue } from "@vue/test-utils"
import TmModalError from "common/TmModalError"

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.directive(`tooltip`, () => {})
localVue.directive(`focus`, () => {})

describe(`TmModalError`, () => {
  let wrapper
  const store = new Vuex.Store({
    getters: {
      lastHeader: () => ({ chain_id: `gaia-test`, height: `31337` }),
    },
  })

  beforeEach(() => {
    wrapper = shallowMount(TmModalError, {
      localVue,
      store,
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has an icon`, () => {
    expect(
      wrapper.find(`.tm-modal-error__icon i.material-icons`).text().trim()
    ).toBe(`error_outline`)
  })

  it(`shows an icon if specified`, () => {
    wrapper.setProps({ icon: `icon-x` })
    expect(
      wrapper.find(`.tm-modal-error__icon i.material-icons`).text().trim()
    ).toBe(`icon-x`)
  })

  it(`has a title`, () => {
    expect(wrapper.find(`.tm-modal-error__title`).text().trim()).toBe(
      `Voyager ran into an error`
    )
  })

  it(`shows a title if specified`, () => {
    wrapper.setProps({ title: `title-x` })
    expect(wrapper.find(`.tm-modal-error__title`).text().trim()).toBe(`title-x`)
  })

  it(`has a body`, () => {
    expect(wrapper.find(`.tm-modal-error__body`).text().trim()).toContain(
      `Voyager has encountered a critical error that blocks the app from running. Please create an issue and include a copy of the app logs.`
    )
  })

  it(`shows a body if specified`, () => {
    wrapper.setProps({ body: `body-x` })
    expect(wrapper.find(`.tm-modal-error__body`).text().trim()).toBe(`body-x`)
  })

  it(`has a button to create an issue`, () => {
    wrapper.find(`#tm-modal-error__btn-issue`).trigger(`click`)
  })
})
