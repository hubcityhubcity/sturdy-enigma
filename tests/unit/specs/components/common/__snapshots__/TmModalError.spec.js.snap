// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmModalError has the expected html structure 1`] = `
<div
  class="tm-modal-error__wrapper"
>
  <div
    class="tm-modal-error"
  >
    <div
      class="tm-modal-error__icon"
    >
      <i
        class="material-icons notranslate"
      >
        error_outline
      </i>
    </div>
     
    <div
      class="tm-modal-error__title"
    >
      
      Voyager ran into an error
    
    </div>
     
    <div
      class="tm-modal-error__body"
    >
      
      Voyager has encountered a critical error that blocks the app from running. Please create an issue and include a copy of the app logs.
    
    </div>
     
    <div
      class="tm-modal-error__footer"
    >
      <tmbtn-stub
        color="primary"
        href="https://github.com/cosmos/voyager/issues"
        icon="bug_report"
        id="tm-modal-error__btn-issue"
        type="anchor"
        value="Create an issue"
      />
    </div>
  </div>
</div>
`;
